### ISSUE TITLE
Entering the background immediately after closing the database causes a crash

### ISSUE THREAD

**Issue by cragod** [groue/GRDB.swift#1612]
After closing the db pool, enter the background immediately. At this time, when performing `releasingMemory`, the `sqliteConnection` is already nil, which will cause a crash at `sqlite3_db_release_memory`

```
Crashed: GRDB.DatabasePool.writer: EXC_BAD_ACCESS KERN_INVALID_ADDRESS 0x0000000000000018
 0                           0x106f22c sqlite3_db_release_memory + 2111308
 1                           0x2414c14 Database.releaseMemory() + 1576 (Database.swift:1576)
 2                          0x2453f54 closure #1 in closure #1 in DatabasePool.releaseMemoryEventually() + 208 (DatabasePool.swift:208)
 3                          0x252a7ec closure #1 in SerializedDatabase.async(_:) + 228 (SerializedDatabase.swift:228)
 4                          0x1cfeec0 thunk for @escaping @callee_guaranteed @Sendable () -> () + 5622064 (<compiler-generated>:5622064)
```

I think we need to check if it is nil.

**Comment by groue** [comment]
Hello @cragod,

Thank you for the report.

I could not reproduce the crash, but I could see an SQLite log message about misuse of `sqlite3_db_release_memory` with a NULL pointer.

I've created #1613 to fix this issue.

**Comment by groue** [comment]
The fix has shipped in v6.29.3
